master/member.go

```diff
@@ -28,25 +31,72 @@ func (m *Member) Unmarshal(data []byte) error {
 	return json.Unmarshal(data, m)
 }
 
-func (s *Server) updateServerMasterMembers(ctx context.Context) error {
-	resp, err := s.etcdClient.MemberList(ctx)
+// Membership defines the interface to query member information in metastore
+type Membership interface {
+	GetConfigs(ctx context.Context) (map[string]*Config, error)
+	GetMembers(ctx context.Context, leader *Member, etcdLeaderID uint64) ([]*Member, error)
+}
+
+type EtcdMembership struct {
+	etcdCli *clientv3.Client
+}
+
+func (em *EtcdMembership) GetConfigs(ctx context.Context) (map[string]*Config, error) {
+	resp, err := em.etcdCli.Get(ctx, adapter.MasterInfoKey.Path(), clientv3.WithPrefix())
```

Comment:
if the membership is a struct ,why not set  _the adapter.MasterInfoKey.Path()__  as a input parameter for this method

Reply:
`MasterInfoKey` and `master.Config` are bounded logic here, so this function doesn't pass a key as parameter.